### PR TITLE
Add e2e coverage of the features doc

### PR DESCRIPTION
**Pull request by Vacant2333**
**What type of PR is this?**
Add features e2e test coverage details
issue:https://github.com/karmada-io/karmada/issues/3685

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: test/e2e/coverage_docs/namespace_test.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+### namespace e2e test coverage analysis
+
+| Id  | Test title                                | Test details                                           |
+|-----|-------------------------------------------|--------------------------------------------------------|
+| 1   | create a namespace in karmada-apiserver   |                                                        |
+| 2   | delete a namespace from karmada-apiserver |                                                        |
+| 3   | joining new cluster                       | namespace should be propagated to new created clusters |
```
Comment (XiShanYongYe-Chang) [inline test/e2e/coverage_docs/namespace_test.md:7]: How about update like this:

```diff
 ### namespace e2e test coverage analysis

-| Id  | Test title                                | Test details                                           |
-|-----|-------------------------------------------|--------------------------------------------------------|
-| 1   | create a namespace in karmada-apiserver   |                                                        |
--- a/test/e2e/coverage_docs/namespace_test.md
+++ b/test/e2e/coverage_docs/namespace_test.md
+
+| Test Case                                                                                                                                      | E2E Describe Text                         | Comments
                                                                                             |
+|------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------|-------------------------------------
---------------------------------------------------------------------------------------------|
+| Create a namespace that needs to be automatically synchronized across all member clusters.                                                     | create a namespace in karmada-apiserver   |
                                                                                             |
+| Delete a namespace, and all member clusters need to automatically synchronize the deletion.                                                    | delete a namespace from karmada-apiserver |
                                                                                             |
+| When a new cluster joins, the namespaces on the Karmada control plane (excluding reserved namespaces) need to be synchronized to that cluster. | joining new cluster                       | [Namespace Management](https://karma
da.io/docs/next/userguide/bestpractices/namespace-management/#default-namespace-propagation) |
~
```

File: test/e2e/coverage_docs/E2E_coverage.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+### E2E coverage of features
```
Comment (XiShanYongYe-Chang) [inline test/e2e/coverage_docs/E2E_coverage.md:1]: I think we can do it after each of our file analyses is completed, then we can analyze the overall characteristics of E2E test coverage.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks a lot~

File: test/e2e/coverage_docs/E2E_coverage.md (hunk 2 above)
Comment (Vacant2333) [reply test/e2e/coverage_docs/E2E_coverage.md:1]: Yes, then I will temporarily keep this file, but delete the current content

File: test/e2e/coverage_docs/namespace_test.md (hunk 1 above)
Comment (Vacant2333) [reply test/e2e/coverage_docs/namespace_test.md:7]: yes, its a good way to show the test cases!

File: test/e2e/coverage_docs/resource_test.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+| JobStatus collection testing: Test the status collection of Job resources.                                                      | JobStatus collection            |                                                                                                                      |
+| DaemonSetStatus collection testing: Test the status collection of DaemonSet resources.                                          | DaemonSetStatus collection      |                                                                                                                      |
+| StatefulSetStatus collection testing: Test the status collection of StatefulSet resources.                                      | StatefulSetStatus collection    |                                                                                                                      |
+| PodDisruptionBudget collection testing: Test the status collection of PodDisruptionBudget resources.                            | PodDisruptionBudget collection  | [Resource Selector](https://karmada.io/docs/next/userguide/globalview/global-search-for-resources/#resourceselector) |
```
Comment (XiShanYongYe-Chang) [inline test/e2e/coverage_docs/resource_test.md:12]: I don't understand why the link about Resource Selector was posted, it doesn't seem to have a direct connection.

Comment (Vacant2333) [reply test/e2e/coverage_docs/resource_test.md:12]: in my view, most of the tests here are based on ResourceSelector.

Comment (XiShanYongYe-Chang) [reply test/e2e/coverage_docs/resource_test.md:12]: I understand what you mean, but we are not using the ResourceSelector of resource `ResourceRegistry`. I think we might need to remove this comment.

Comment (Vacant2333) [reply test/e2e/coverage_docs/resource_test.md:12]: ok, i understand

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Good job! We can analyze the other files in this way. 
If it is found later that the name or quantity of the list item is inappropriate or cannot represent the current situation, adjustments can be made at any time.
/lgtm
/approve

**Pull request merged**